src/Components/Article.jsx

```diff
@@ -0,0 +1,52 @@
+import {getArticleById} from "../utils/getArticleById.js";
+import {useEffect, useState} from "react";
+import {useParams} from "react-router-dom";
+import {Button, Card, ListGroup} from "react-bootstrap";
+
+export const Article = () => {
+  const [article, setArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  let { id } = useParams();
+
+  useEffect(() => {
+    const fetchArticle = async () => {
+      try {
+        const article = await getArticleById(id);
+        setArticle(article);
+      } catch (error) {
+        console.log(error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+
+    if (isLoading) {
+      <h1>Loading...</h1>
+    }
+
+    fetchArticle();

```

Comment:
i would argue you dont need to declare the function here to use it here if it's not going to be used anywhere else? You can have the try catch block directly in the useEffect